front/pages/api/v1/data_sources/[user]/[name]/search.ts

```diff
@@ -0,0 +1,230 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { auth_api_user } from "@app/lib/api/auth";
+import { APIError, APIErrorWithStatusCode } from "@app/lib/api/error";
+import { Result, Ok, Err } from "@app/lib/result";
+import { JSONSchemaType } from "ajv";
+import { User, DataSource, Provider } from "@app/lib/models";
+import { Op } from "sequelize";
+import { credentialsFromProviders } from "@app/lib/providers";
+import { parse_payload } from "@app/lib/http_utils";
+
+type TagsFilter = {
+  in?: string[];
+  not?: string[];
+};
+
+type TimestampFilter = {
+  gt?: number;
+  lt?: number;
+};
+
+type SearchFilter = {
+  tags?: TagsFilter;
+  timestamp?: TimestampFilter;
+};
+
+export type DatasourceSearchQuery = {
+  query: string;
+  top_k: number;
+  full_text: boolean;
+  timestamp_gt?: number;
+  timestamp_lt?: number;
+  tags_in?: string[];
+  tags_not?: string[];
+};
+
+const searchQuerySchema: JSONSchemaType<DatasourceSearchQuery> = {
+  type: "object",
+  properties: {
+    query: { type: "string" },
+    top_k: { type: "number" },
+    full_text: { type: "boolean" },
+    timestamp_gt: { type: "number", nullable: true },
+    timestamp_lt: { type: "number", nullable: true },
+    tags_in: { type: "array", items: { type: "string" }, nullable: true },
+    tags_not: { type: "array", items: { type: "string" }, nullable: true },
+  },
+  required: ["query", "top_k", "full_text"],
+};
+
+const { DUST_API } = process.env;
+
+type DatasourceSearchResponseBody = {
+  documents: Array<Document>;
+};
+
+export async function performSearch(
```

Comment:
Let's remove the function call here as well and move the code directly in the handler out of consistency with the rest of the codebase :+1: